src/main/groovy/wooga/gradle/snyk/tasks/SnykInstall.groovy

```diff
@@ -14,21 +16,45 @@ import wooga.gradle.snyk.tasks.internal.SnykDownloadAsset
 
 class SnykInstall extends DefaultTask implements SnykInstallSpec {
 
-    private static final String SNYK_RELEASE_JSON_URL = "https://static.snyk.io/cli/%s/release.json"
+    static String getReleaseJsonUrl(String version) {
+        def scheme = System.getProperty("static.snyk.io.scheme", "https")
+        def port = System.getProperty("static.snyk.io.port", "443")
+        def host = System.getProperty("static.snyk.io.host", "static.snyk.io")
+        "${scheme}://${host}:${port}/cli/${version}/release.json"
+    }
 
     @OutputFile
     final Provider<RegularFile> snykExecutable
 
     private final Provider<Map<String, SnykDownloadAsset>> snykReleasesProvider
 
-    SnykInstall() {
-        this.snykReleasesProvider =  version.map ({
-            String version -> fetchSnykReleases(version)
-        }.memoize())
+    private final Provider<String> versionToInstall
+
+    @Input
+    Provider<String> getVersionToInstall() {
+        versionToInstall
+    }
 
+    SnykInstall() {
         this.snykExecutable = installationDir.zip(executableName) {
             dir, fileName -> dir.file(normalizeExecutableByOS(fileName))
         }
+
+        this.versionToInstall = version.map({
+            def version = it
+            if (version == "latest") {
+                def releasesURL = getReleaseJsonUrl(version)
+                def result = new JsonSlurper().parse(new URL(releasesURL)) as Map
+                version = result.version
+            } else if (version.startsWith("v")) {
+                version = version.substring(1)
```

Comment:
Why do we get rid of the "v" just to put it back in the `fetchSynkReleases` method?

Reply:
Because snyk uses the scheme `v0.0.0` only for the URL. In the release json and in the cli with `--version` they report it as as simple semver version. I made the change to work without the `v` when setting the version. In the end it won't really matter. The only difference will be the directory name where snyk will be installed. I just made sure that one can set it without the `v` and that we track the property `versionToInstall` in the same normalized form.

Reply:
Ok second reason. `latest` gets converted to a version without a `v` so I wanted to make sure that this map creates always version with the same pattern.

Reply:
👍